codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -141,6 +141,17 @@ public void close() {
         frameReader.close();
     }
 
+    /**
+     * Calculate the threshold in bytes which should trigger a {@code GO_AWAY} if a set of headers exceeds this amount.
+     * @param maxHeaderListSize
+     *      <a href="https://tools.ietf.org/html/rfc7540#section-6.5.2">SETTINGS_MAX_HEADER_LIST_SIZE</a> for the local
+     *      endpoint.
+     * @return the threshold in bytes which should trigger a {@code GO_AWAY} if a set of headers exceeds this amount.
+     */
+    protected long maxHeaderListSizeGoAway(long maxHeaderListSize) {
+        return maxHeaderListSize + (maxHeaderListSize >>> 2);
```

Comment:
maybe just `1.5 * maxHeaderListSize` to make it more immediately clear? Also, do you need to take into account overflow? I guess it's a long so maybe not?